blog/content/post/2024-03-15-uefi-sb-implementation-mpl.md

```diff
@@ -0,0 +1,385 @@
+
+```bash
+robot -L TRACE \
+  -v ansible_config:yes \
+  -v rte_ip:<rte_ip> \
+  -v snipeit:no \
+  -v config:mpl-pip4 \
+  -v device_ip:<device_ip> \
+  dasharo-security/secure-boot.robot
+```
+
+To use QEMU execute this instead:
+
+```bash
+robot -L TRACE \
+  -v ansible_config:yes \
+  -v rte_ip:127.0.0.1 \
+  -v snipeit:no \
+  -v config:qemu \
+  -v device_ip:<device_ip> \
+  dasharo-security/secure-boot.robot
+```
+
+### Results
+
+The tests were performed on MPL PIP44. The following images show the test suite
+results:
+![uefi-sb-results-report](/img/uefi-sb-results-pt1.png)
+![uefi-sb-results-report](/img/uefi-sb-results-pt2.png)
+
+The results lead to the following conclusions:
+
+- The state of UEFI Secure Boot functionality can be freely modified from the
+UEFI BIOS Menu, and it is correctly detected from the operating system.
+- Verification of launched images works correctly when UEFI Secure Boot is
+enabled. The firmware allows the execution of files signed with the appropriate
+keys but blocks the booting of unsigned files or files signed with keys not
+present in the database (DB) or files with hashes not stored in the database
+(DB).
+- The firmware correctly recognizes the chain of trust when intermediate
+certificates are used, allowing their use in the verification process.
+- Keys intended for UEFI Secure Boot must be generated using the RSA
+cryptographic algorithm and sizes 2048, 3072, and 4096. Keys generated with the
+ECDSA cryptographic algorithm are not correctly supported.
+- The firmware does not verify the expiration date of certificates during the
+verification of launched files. It means that enrolled certificates may expire
+and will not affect the ability to boot files verified by them.
+- The firmware only allows the resetting of enrolled certificates when they
+have been added from the UEFI BIOS Menu. If they are added from the operating
+system, the certificates are marked as External, and only a full firmware
+reset (for example, by removing the CMOS battery) allows their removal.
+- Certificates enrolled through the Automatic Certificate Provisioning method
+are correctly used to verify launched files.
+- The sbctl tool can be used to manage UEFI Secure Boot certificates.
+- Automatic tests of the sbctl tool and the Automatic Certificate Provisioning
+methods have been omitted due to their logic, assuming that firmware can remove
+certificates enrolled this way from the UEFI BIOS Menu.
+
+The conclusions allowed us to proceed with Secure Boot integration in the
+Yocto layer.
+
+## Integrating Secure Boot into a Yocto layer
+
+Integrating UEFI Secure Boot into an existing Yocto layer is possible by using
+the [meta-secure-core](https://github.com/Wind-River/meta-secure-core) layer in
+your build. Its sublayer – `meta-efi-secure-boot` introduces mechanisms that
+allow verifying various files used in the boot process. It offers two secure
+boot technologies:
```

Comment:
Don't we have our own fork with modifications?

Reply:
We used this one to implemement SB.